scrapy/exporters.py

```diff
@@ -252,16 +261,28 @@ def _build_row(self, values):
 
     def _write_headers_and_set_fields_to_export(self, item):
         if self.include_headers_line:
+            fields = self._get_fields_from_item(item)
             if not self.fields_to_export:
```

Comment:
I'm not sure this is correct. Unless I'm missing something, this will show a warning if `FEED_EXPORT_FIELDS` was not set even if all items have the same fields, and hence all fields are exported. Could you add a test case for this scenario?

Reply:
Yes, this was originally the intention that if the `FEED_EXPORT_FIELDS `is not set, then we would show that there could be a chance of a _**possible**_ data loss. But subsequently it was decided we would not tell _possible_  but will tell that an actual data loss is detected when the fields length differ.

Reply:
@elacuesta should I keep this logic? or change the warning message to `"Possible data loss detected when exporting items -- This message won't be shown for further items"` if the `FEED_EXPORT_FIELDS` was not set even if all the items have the same fields.